pkg/provisioner/ironic/ironic.go

```diff
@@ -1065,18 +1065,22 @@ func (p *ironicProvisioner) Adopt(data provisioner.AdoptData, force bool) (resul
 	return operationComplete()
 }
 
-func (p *ironicProvisioner) ironicHasSameImage(ironicNode *nodes.Node, image metal3v1alpha1.Image) (sameImage bool) {
+func (p *ironicProvisioner) ironicHasSameImage(ironicNode *nodes.Node, image *metal3v1alpha1.Image) (sameImage bool) {
 	// To make it easier to test if ironic is configured with
 	// the same image we are trying to provision to the host.
-	if image.DiskFormat != nil && *image.DiskFormat == "live-iso" {
+	if image != nil && image.DiskFormat != nil && *image.DiskFormat == "live-iso" {
```

Comment:
This is correct insofar as it will prevent anything from crashing, but is this the logic we actually want?
If the image is nil, shouldn't we check that none of the image fields are set in ironic?